builtin/providers/google/provider.go

```diff
@@ -270,17 +270,20 @@ func getNetworkLink(d *schema.ResourceData, config *Config, field string) (strin
 func getNetworkName(d *schema.ResourceData, field string) (string, error) {
 	if v, ok := d.GetOk(field); ok {
 		network := v.(string)
+		return getNetworkNameFromString(network)
+	}
+	return "", nil
+}
 
-		if strings.HasPrefix(network, "https://www.googleapis.com/compute/") {
-			// extract the network name from SelfLink URL
-			networkName := network[strings.LastIndex(network, "/")+1:]
-			if networkName == "" {
-				return "", fmt.Errorf("network url not valid")
-			}
-			return networkName, nil
+func getNetworkNameFromString(network string) (string, error) {
```

Comment:
Nit: Would it be more helpful to call this `getNetworkNameFromSelfLink` or something similar? The `string` part is implied by the type signature, but what's in the string isn't apparent at first glance.